### DISCUSSION TITLE
Keyrah2: RAmiga and Help keys are not working

### DISCUSSION THREAD

**Discussion by ninjaweb62** [BlitterStudio/amiberry#878] [Q&A] [answered]
I'm trying to understand if it's a problem of mine or keyrah2's one.
I'm using keyrah2 with a A1200 keyboard and and RPi4. I use last Amiberry 4.2.0 compiled on RaspiOs bullseye against SDL2 v2.0.12.
It looks like Right Amiga and Help key are non working. I.e i can't use RAMIGA/I to open an icon info widows.
Tested the keyboard with Amiga Test Kit, here are the results:
Right Amiga: no raw keycode
Help: shows 66 e6 raw keycode (just the same as Left Amiga)
All other keys are correctly working.
Any suggestion ?

Thanks

**Comment by midwan** [comment, marked as answer]
@ninjaweb62 
It's hard to say, as I don't have a device to test with here and try to recreate this.
The relevant code came from WinUAE so it _should_ work, but I don't know if there's something wrong without being able to test it all the way. :)

**Reply by midwan** [reply]
Have you tried different configurations, as those change the mappings, based on the documentation seen here? http://wiki.icomp.de/wiki/Keyrah_V2

**Reply by ninjaweb62** [reply]
Thanks midwan, obviouly I already tried that, but that remapping is only useful for A600 keyboard which misses the numeric key pad. According to the doc, Amiga-keys should always become Windows-keys, and HELP becomes "insert" key code.
Is there any file in the Amiberry sources I can investigate to remap a key ?
I forgot to say that months ago the Amiga keys were working fine!
I think I should try with windows WinUAE and check how keyrah behaves...

**Reply by midwan** [reply]
The files `src/osdep/keyboard.cpp` and `src/osdep/amiberry_input.cpp` are relevant here.
The first one has the general keyboard mapping, and as you can see there, Windows keys are mapped to the Amiga keys by default. There is an exception here, for keyboards that don't have a right Windows key: we have an option to map Right Ctrl as the Right Amiga key, which you can enable from the GUI (in the Misc panel) or enable it by default in `amiberry.conf` if you want.

The HELP key is normally mapped to PageDown, and the Insert key maps to LAmiga - which is what you're seeing when testing.

**Reply by midwan** [reply]
So from what I can deduce so far, it seems that:
- the Help -> Insert is expected with our current mapping, so that would require some change to support keyrah v2
- The RAmiga key not triggering anything is a bit weird, as I would expect it to trigger _something_. That would require some troubleshooting on the SDL2 level, to see what kind of raw input we get when that is pressed. If it's just some key that is not mapped, that's again something that could be fixed with some modifications.
- 
Maybe some option to enable Keyrah V2 mappings could be added, if we figure out what the needed changes are. :)

**Reply by ninjaweb62** [reply]
I tried under Windows WinUAE. The behaviour was exactly the same as Amiberry, so I immediately dismounted the keyboard and discovered that a few membrane tracks looks a little bit damaged (some strange liquid on them).
I already ordered a new menbrane.
Thanks a lot midwand, sorry if I made you loose you some time. If hope to fix the keyboard soon.

Happy new year !!!
